core/src/components/datetime/utils/format.ts

```diff
@@ -183,11 +183,25 @@ export const getYear = (locale: string, refParts: DatetimeParts) => {
   return getLocalizedDateTime(locale, refParts, { year: 'numeric' });
 };
 
-const getNormalizedDate = (refParts: DatetimeParts) => {
+/**
+ * Given reference parts, return a JS Date object
+ * with a normalized time.
+ */
+export const getNormalizedDate = (refParts: DatetimeParts) => {
   const timeString =
     refParts.hour !== undefined && refParts.minute !== undefined ? ` ${refParts.hour}:${refParts.minute}` : '';
 
-  return new Date(`${refParts.month}/${refParts.day}/${refParts.year}${timeString} GMT+0000`);
+  /**
+   * We use / notation here for the date
+   * so we do not need to do extra work and pad values with zeroes.
+   * Values such as YYYY-MM are still valid, so
+   * we add fallback values so we still get
+   * a valid date otherwise we will pass in a string
+   * like "//2023". Some browsers, such as Chrome, will
+   * account for this and still return a valid date. However,
+   * this is not a consistent behavior across all browsers.
+   */
+  return new Date(`${refParts.month ?? 1}/${refParts.day ?? 1}/${refParts.year ?? 2023}${timeString} GMT+0000`);
```

Comment:
Should we fallback to the current year, instead of hardcoding 2023?

e.g.:
```ts
refParts.year ?? new Date().getFullYear()
```

Reply:
We could, though we also default to 1/1/2023 here: https://github.com/ionic-team/ionic-framework/blob/89ed213a582877f45f91a77f3cc3905715114384/core/src/components/datetime/utils/format.ts#L58

Any preference?

Reply:
> Presumably, if I'm passing in a value without the year, I'm using the month presentation, in which case I don't care about the year since it'll never be displayed.

Yep exactly this. If any of those values are `undefined` then they aren't actually used in the rendering of the component -- we just need them to get a valid date

Reply:
I'm ok with this as-is then 👍